### PR TITLE
Private AKS Cluster

### PR DESCRIPTION
**Pull request by paolosalvatori**
# PR Checklist

This ARM template includes all the latest features like private AKS clusters, new simplified AKS-managed AAD integration, the brand new Azure RBAC for Kubernetes Authorization, actually in preview, and the use of managed identity in place of a service principal, and more.

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* First Commit

### CODE REVIEW

File: 201-private-aks-cluster/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,1031 @@
...
+    },
+    "aksClusterKubernetesVersion": {
+      "type": "string",
+      "defaultValue": "1.17.9",
+      "metadata": {
+        "description": "Specifies the version of Kubernetes specified when creating the managed cluster."
+      }
+    },
+    "aksClusterAdminUsername": {
+      "defaultValue": "azureuser",
```
Comment (bmoore-msft) [inline 201-private-aks-cluster/azuredeploy.json:116]: remove defaultValue

File: 201-private-aks-cluster/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,1031 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "Specifies the administrator username of Linux virtual machines."
+      }
+    },
+    "aksClusterSshPublicKey": {
+      "defaultValue": null,
```
Comment (bmoore-msft) [inline 201-private-aks-cluster/azuredeploy.json:123]: remove defaultValue

File: 201-private-aks-cluster/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,1031 @@
...
+    "vmAdminPasswordOrKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Specifies the SSH Key or password for the virtual machine. SSH key is recommended."
+      }
+    },
+    "diskStorageAccounType": {
+      "type": "string",
+      "defaultValue": "Premium_LRS",
+      "allowedValues": [
```
Comment (bmoore-msft) [inline 201-private-aks-cluster/azuredeploy.json:421]: See https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate#storageaccounttypes for supported types for disks

File: 201-private-aks-cluster/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,1031 @@
...
+      }
+    },
+    "dataDiskCaching": {
+      "type": "string",
+      "defaultValue": "ReadWrite",
+      "metadata": {
+        "description": "Specifies the caching requirements for the data disks."
+      }
+    },
+    "blobStorageAccountName": {
```
Comment (bmoore-msft) [inline 201-private-aks-cluster/azuredeploy.json:465]: boot? or blob?

File: 201-private-aks-cluster/azuredeploy.parameters.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
```
Comment (bmoore-msft) [inline 201-private-aks-cluster/azuredeploy.parameters.json:4]: remove params for which you have defaultValues so we can test those...

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 201-private-aks-cluster/azuredeploy.json (hunk 4 above)
Comment (paolosalvatori) [reply 201-private-aks-cluster/azuredeploy.json:465]: This blob storage is used for boot diagnostics logs. Said that I changed its default name as follows:

"blobStorageAccountName": {
      "type": "string",
      "defaultValue": "[concat('blob', uniquestring(resourceGroup().id))]",
      "metadata": {
        "description": "Specifies the globally unique name for the storage account used to store the boot diagnostics logs of the virtual machine."
      }
    },

File: 201-private-aks-cluster/azuredeploy.json (hunk 3 above)
Comment (paolosalvatori) [reply 201-private-aks-cluster/azuredeploy.json:421]: Done

File: 201-private-aks-cluster/azuredeploy.json (hunk 2 above)
Comment (paolosalvatori) [reply 201-private-aks-cluster/azuredeploy.json:123]: Done

File: 201-private-aks-cluster/azuredeploy.json (hunk 1 above)
Comment (paolosalvatori) [reply 201-private-aks-cluster/azuredeploy.json:116]: Done

File: 201-private-aks-cluster/azuredeploy.parameters.json (hunk 5 above)
Comment (paolosalvatori) [reply 201-private-aks-cluster/azuredeploy.parameters.json:4]: Done!

**Comment by bmoore-msft** [comment]
nit - can you format the json in azuredeploy.json - otherwise that's going to create diffs we don't care about (but still have to look at) like it did for this one...  I'll merge this but pls do another one quickly...

**Pull request merged**